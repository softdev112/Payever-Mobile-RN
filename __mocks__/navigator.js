export default {
  setOnNavigatorEvent: jest.fn(),
  resetTo: jest.fn(),
  push: jest.fn(),
  dismissLightBox: jest.fn(),
};