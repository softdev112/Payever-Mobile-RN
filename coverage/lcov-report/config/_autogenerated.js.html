<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for config/_autogenerated.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">config</a> _autogenerated.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/* eslint-disable max-len */
export default {
  siteUrl: 'https://stage.payever.de',
  api: {
    baseUrl: 'https://stage.payever.de',
    clientId: '117_5sd54ii2lyo8sg84g0w4w08wkw4040gks4k0sk0wgc44w48404',
    clientSecret: '4jsu5lynawsg8g4cs80w4kcsocwcggwo048ogs8c844gk80w80',
  },
&nbsp;
  oauthData: {
    twitter: {
      appId: 'oRCkHDAnAFSXiO1wayGlpBkwn',
      appSecret: 'C3FUgIBnjdn1Je6dD5NX5AboFEJ1vWqVVwOLJ03Itk0Xp58but',
      callback: '/social/auth/twitter',
    },
&nbsp;
    facebook: {
      appId: '306625799760428',
      appSecret: '4e0e5079b5c7eeff9f16295f0212ca95',
      callback: 'fb306625799760428://authorize',
      scope: 'email public_profile',
      fields: [
        'id', 'name', 'first_name', 'last_name',
        'picture.type(large)', 'email', 'cover', 'birthday',
      ],
    },
&nbsp;
    google: {
      appId: '463919467171-u54vacd9oiso7l14f97g38odsak91njg.apps.googleusercontent.com',
      appSecret: 'SihRvMNWT2j5088wEZ_A2MRR',
      callback: 'com.googleusercontent.apps.463919467171-u54vacd9oiso7l14f97g38odsak91njg:/google',
      scope: 'https://www.googleapis.com/auth/content',
    },
&nbsp;
    linkedIn: {
      appId: '78a1v96xm7x1tt',
      appSecret: 'Umu6R25x1w7NiDt6',
      callback: 'https://localhost:3003/auth/linkedin',
      scope: 'r_basicprofile r_emailaddress',
    },
  },
};</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Jul 06 2017 09:31:11 GMT+0300 (MSK)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
