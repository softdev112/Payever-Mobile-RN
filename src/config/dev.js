/* eslint-disable max-len */
import local from './local';

export default Object.assign({
  siteUrl: 'https://mein.payever.de',
  api: {
    baseUrl: 'https://mein.payever.de',
    clientId: '1633_3vi6g4uiwmyow8044o4w0wo4s88cogosw84kw888kw408wok8c',
    clientSecret: '63nqx0kppzocw4sswok8gg800sk4w4kko4k0oo8gs4ow00wgos',
  },
  debug: {
    logApiCall: true,
  },

  oauthData: {
    twitter: {
      appId: 'oRCkHDAnAFSXiO1wayGlpBkwn',
      appSecret: 'C3FUgIBnjdn1Je6dD5NX5AboFEJ1vWqVVwOLJ03Itk0Xp58but',
      callback: '/social/auth/twitter',
    },

    facebook: {
      appId: '438908279593357',
      appSecret: 'cd96a62e92a5bc498896209613e2d0bf',
      callback: '/social/auth/facebook',
      scope: 'email public_profile',
      fields: [
        'id', 'name', 'first_name', 'last_name',
        'picture.type(large)', 'email', 'cover', 'birthday',
      ],
    },

    google: {
      appId: '463919467171-u54vacd9oiso7l14f97g38odsak91njg.apps.googleusercontent.com',
      appSecret: 'SihRvMNWT2j5088wEZ_A2MRR',
      callback: 'com.googleusercontent.apps.463919467171-u54vacd9oiso7l14f97g38odsak91njg:/google',
      scope: 'https://www.googleapis.com/auth/content',
    },

    linkedIn: {
      appId: '78a1v96xm7x1tt',
      appSecret: 'Umu6R25x1w7NiDt6',
      callback: 'https://localhost:3003/auth/linkedin',
      scope: 'r_basicprofile r_emailaddress',
    },
  },
}, local);