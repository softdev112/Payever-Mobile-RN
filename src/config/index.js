import generatedConfig from './_autogenerated';

export class Config {
  siteUrl: string;
  api: {
    baseUrl: string;
    clientId: string;
    clientSecret: string;
  };
  debug: {
    consoleLevel: boolean;
    logApiCall: boolean;
    logMobx: boolean;
    logSLevel: boolean;
    logSUrl: string;
    logWampCall: boolean;
  };

  constructor(data) {
    data.debug = data.debug || {};
    Object.assign(this, data);
  }
}

export default new Config(generatedConfig);