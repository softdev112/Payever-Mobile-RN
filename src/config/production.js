/* eslint-disable max-len */
export default {
  siteUrl: 'https://mein.payever.de',
  // It's better to prevent using clientSecret. But after analyzing a lot
  // of resources I've made a decision that is's impossible.
  // FOSOAuthServerBundle doesn't support password auth without client_secret
  // https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/issues/115
  api: {
    baseUrl: 'https://mein.payever.de',
    clientId: '1633_3vi6g4uiwmyow8044o4w0wo4s88cogosw84kw888kw408wok8c',
    clientSecret: '63nqx0kppzocw4sswok8gg800sk4w4kko4k0oo8gs4ow00wgos',
  },
  reqTimeout: 10000,

  oauthData: {
    twitter: {
      appId: 'oRCkHDAnAFSXiO1wayGlpBkwn',
      appSecret: 'C3FUgIBnjdn1Je6dD5NX5AboFEJ1vWqVVwOLJ03Itk0Xp58but',
      callback: '/social/auth/twitter',
    },

    facebook: {
      appId: '302838276848223',
      appSecret: '188dcf1590b667929d5f8b4538b00075',
      callback: 'fb302838276848223://authorize',
      scope: 'email public_profile',
      fields: [
        'id', 'name', 'first_name', 'last_name',
        'picture.type(large)', 'email', 'cover', 'birthday',
      ],
    },

    google: {
      appId: '463919467171-u54vacd9oiso7l14f97g38odsak91njg.apps.googleusercontent.com',
      appSecret: 'SihRvMNWT2j5088wEZ_A2MRR',
      callback: 'com.googleusercontent.apps.463919467171-u54vacd9oiso7l14f97g38odsak91njg:/google',
      scope: 'https://www.googleapis.com/auth/content',
    },

    linkedIn: {
      appId: '78a1v96xm7x1tt',
      appSecret: 'Umu6R25x1w7NiDt6',
      callback: 'https://localhost:3003/auth/linkedin',
      scope: 'r_basicprofile r_emailaddress',
    },
  },
};