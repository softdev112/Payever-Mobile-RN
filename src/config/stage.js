export default {
  siteUrl: 'https://stage.payever.de',
  api: {
    baseUrl: 'https://stage.payever.de',
    clientId: '117_5sd54ii2lyo8sg84g0w4w08wkw4040gks4k0sk0wgc44w48404',
    clientSecret: '4jsu5lynawsg8g4cs80w4kcsocwcggwo048ogs8c844gk80w80'
  }
}