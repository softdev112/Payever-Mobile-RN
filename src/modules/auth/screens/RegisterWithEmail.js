import { Component } from 'react';
import { inject, observer } from 'mobx-react/native';
import { findNodeHandle, Keyboard, ScrollView } from 'react-native';
import { FlatTextInput, Loader, NavBar, StyleSheet, ValidationInformer, View,
} from 'ui';
import { Navigator } from 'react-native-navigation';

import AuthStore from '../../../store/auth';

const KEYBOARD_SCROLL_ADJUST = 80;
const FORM_ELEMENTS_COUNT = 5;

const EMAIL_VALID_PATTERN =
  '^([a-z0-9_-]+\\.)*[a-z0-9_-]+@[a-z0-9_-]+(\\.[a-z0-9_-]+)*\\.[a-z]{2,6}$';

/**
 * Form validation messages
 * 0 - First Name
 * 1 - Last Name
 * 2 - Email Address
 * 3 - Password
 * 4 - Confirm Password
 */
/* eslint-disable max-len */
const neutralMessages = [];
neutralMessages[0] = 'Enter your first name please';
neutralMessages[1] = 'Enter your last name please';
neutralMessages[2] = 'Enter your email address please';
neutralMessages[3] = 'Choose and enter your password';
neutralMessages[4] = 'Confirm your password please';

const errorMessages = [];
errorMessages[0] = 'It can\'t be empty. Enter your first name please';
errorMessages[1] = 'It can\'t be empty. Enter your last name please';
errorMessages[2] = 'It should be not empty valid email address for example someone@gmail.de';
errorMessages[3] = 'It can\'t be empty. Choose your password please';
errorMessages[4] = 'Should match with password. Try to enter it again';

const successMessages = [];
successMessages[0] = 'Your first name is:';
successMessages[1] = 'Your last name is:';
successMessages[2] = 'Your email is:';
successMessages[3] = 'Your password is:';
successMessages[4] = 'Your password is confirmed:';
/* eslint-enable max-len */

@inject('auth')
@observer
export default class RegisterWithEmail extends Component {
  static navigatorStyle = {
    navBarHidden: true,
  };

  props: {
    auth: AuthStore;
    navigator: Navigator;
  };

  state: {
    formElementsData: Array<string>;
    inputNodeId: number;
    isSaveAttempt: boolean;
  };

  /**
   * Form validation messages
   * 0 - First Name
   * 1 - Last Name
   * 2 - Email Address
   * 3 - Password
   * 4 - Confirm Password
   */
  $formInputs: Array<FlatTextInput> = [];
  $scrollView: ScrollView;

  constructor(props) {
    super(props);

    this.state = {
      formElementsData: Array(FORM_ELEMENTS_COUNT).fill(''),
      inputNodeId: -1,
      isSaveAttempt: false,
    };
  }

  async onCreateNewUser() {
    const { auth, navigator } = this.props;
    const { formElementsData } = this.state;

    this.setState({ isSaveAttempt: true });

    const notValidIdx = formElementsData.findIndex(element => element === '');

    if ((notValidIdx === -1 || notValidIdx > 2)
      && !formElementsData[2].match(EMAIL_VALID_PATTERN)) {
      if (this.$formInputs[2]) {
        this.$formInputs[2].shakeElementAndSetFocus();
      }
      return;
    }

    if (notValidIdx !== -1) {
      if (this.$formInputs[notValidIdx]) {
        this.$formInputs[notValidIdx].shakeElementAndSetFocus();
      }
      return;
    }

    if (formElementsData[3] !== formElementsData[4]) {
      if (this.$formInputs[4]) {
        this.$formInputs[4].shakeElementAndSetFocus();
      }
      return;
    }

    const newUser = {
      email: formElementsData[2],
      plainPassword: {
        first: formElementsData[3],
        second: formElementsData[4],
      },
      billingAddress: {
        firstName: formElementsData[0],
        lastName: formElementsData[1],
      },
    };

    Keyboard.dismiss();
    await auth.registerNewUser(newUser);

    navigator.showLightBox({
      screen: 'auth.RegCompletedDialog',
      animated: true,
      style: {
        backgroundBlur: 'dark',
      },
    });
  }

  onInputInFocus({ nativeEvent }) {
    if (!nativeEvent || !nativeEvent.target) return;

    setTimeout(() => {
      const scrollResponder = this.$scrollView.getScrollResponder();
      scrollResponder.scrollResponderScrollNativeHandleToKeyboard(
        findNodeHandle(nativeEvent.target),
        KEYBOARD_SCROLL_ADJUST,
        true
      );
    }, 50);

    this.setState({ inputNodeId: nativeEvent.target });
  }

  onInputBlur({ nativeEvent }) {
    if (!nativeEvent || !nativeEvent.target) return;

    setTimeout(() => {
      // Scroll back only if there are no inputs in focus
      if (this.state.inputNodeId === -1) {
        const scrollResponder = this.$scrollView.getScrollResponder();
        scrollResponder.scrollResponderScrollNativeHandleToKeyboard(
          findNodeHandle(nativeEvent.target),
          0,
          true
        );
      }
    }, 50);

    this.setState({ inputNodeId: -1 });
  }

  onInputValueChange(index, value) {
    const { formElementsData } = this.state;
    formElementsData[index] = value;
    this.setState({ formElementsData, isSaveAttempt: false });
  }

  onSubmitEditing(index: number) {
    if (index < FORM_ELEMENTS_COUNT - 1 && this.$formInputs[index + 1]) {
      this.$formInputs[index + 1].setFocus();
    }
  }

  renderFieldValidator(fieldIndex: number, options: ValidationOptions = {}) {
    const { formElementsData, isSaveAttempt } = this.state;
    let isFieldValid = formElementsData[fieldIndex] !== '';

    if (options.isEmail) {
      isFieldValid = formElementsData[fieldIndex].match(EMAIL_VALID_PATTERN);
    }

    const matchIndex = options.shouldMatchField;
    if (matchIndex !== undefined) {
      isFieldValid = formElementsData[fieldIndex] !== '' &&
        formElementsData[fieldIndex] === formElementsData[matchIndex];
    }

    let validState = 'neutral';
    if (isSaveAttempt) {
      if (isFieldValid) {
        validState = 'success';
      } else {
        validState = 'error';
      }
    } else if (isFieldValid) {
      validState = 'success';
    }

    return (
      <ValidationInformer
        neutralMessage={neutralMessages[fieldIndex]}
        errorMessage={errorMessages[fieldIndex]}
        successMessage={successMessages[fieldIndex]}
        currentState={validState}
      />
    );
  }

  render() {
    const { auth } = this.props;
    const { formElementsData } = this.state;

    return (
      <View style={styles.container}>
        <NavBar popup>
          <NavBar.Back />
          <NavBar.Title showTitle="always" title="Add New User" />
          <NavBar.Button title="Sign Up" onPress={::this.onCreateNewUser} />
        </NavBar>
        <Loader isLoading={auth.isLoading}>
          <ScrollView
            scrollEnabled={false}
            contentContainerStyle={styles.content}
            ref={ref => this.$scrollView = ref}
          >
            <FlatTextInput
              inputStyle={styles.fieldsText}
              ref={ref => this.$formInputs[0] = ref}
              placeholder="First Name"
              onChangeText={text => this.onInputValueChange(0, text)}
              onFocus={::this.onInputInFocus}
              onBlur={::this.onInputBlur}
              value={formElementsData[0]}
              renderValidator={() => this.renderFieldValidator(0)}
              onSubmitEditing={() => this.onSubmitEditing(0)}
              returnKeyType="next"
            />

            <FlatTextInput
              inputStyle={styles.fieldsText}
              ref={ref => this.$formInputs[1] = ref}
              placeholder="Last Name"
              onChangeText={text => this.onInputValueChange(1, text)}
              onFocus={::this.onInputInFocus}
              onBlur={::this.onInputBlur}
              value={formElementsData[1]}
              renderValidator={() => this.renderFieldValidator(1)}
              onSubmitEditing={() => this.onSubmitEditing(1)}
              returnKeyType="next"
            />

            <FlatTextInput
              inputStyle={styles.fieldsText}
              ref={ref => this.$formInputs[2] = ref}
              placeholder="Email Address"
              onChangeText={text => this.onInputValueChange(2, text)}
              onFocus={::this.onInputInFocus}
              onBlur={::this.onInputBlur}
              value={formElementsData[2]}
              renderValidator={
                () => this.renderFieldValidator(2, { isEmail: true })
              }
              onSubmitEditing={() => this.onSubmitEditing(2)}
              returnKeyType="next"
            />

            <FlatTextInput
              inputStyle={styles.fieldsText}
              secureTextEntry
              ref={ref => this.$formInputs[3] = ref}
              placeholder="Password"
              onChangeText={text => this.onInputValueChange(3, text)}
              onFocus={::this.onInputInFocus}
              onBlur={::this.onInputBlur}
              renderValidator={() => this.renderFieldValidator(3)}
              onSubmitEditing={() => this.onSubmitEditing(3)}
              returnKeyType="next"
            />

            <FlatTextInput
              inputStyle={styles.fieldsText}
              secureTextEntry
              ref={ref => this.$formInputs[4] = ref}
              placeholder="Confirm Password"
              onChangeText={text => this.onInputValueChange(4, text)}
              onFocus={::this.onInputInFocus}
              onBlur={::this.onInputBlur}
              renderValidator={
                () => this.renderFieldValidator(4, { shouldMatchField: 3 })
              }
              onSubmitEditing={() => {}}
              returnKeyType="done"
            />
          </ScrollView>
        </Loader>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },

  content: {
    paddingHorizontal: 12,
    justifyContent: 'flex-start',
    overflow: 'hidden',
  },

  fieldsText: {
    fontSize: 18,
  },
});

type ValidationOptions = {
  isEmail: boolean;
  shouldMatchField: number;
};