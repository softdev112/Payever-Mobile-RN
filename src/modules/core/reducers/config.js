import { Map } from 'immutable';
import config from '../../../config';

const initialState = Map(config);

/**
 * Share config through store
 */
export default function configReducer(state = initialState) {
  return state;
}