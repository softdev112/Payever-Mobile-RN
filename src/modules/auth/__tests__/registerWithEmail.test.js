/* eslint-disable max-len */
import React from 'react';
import renderer from 'react-test-renderer';
import RegisterWithEmail from '../screens/RegisterWithEmail';
import Store from '../../../store';
import config from '../../../config';
import navigator from '../../../../__mocks__/navigator';

jest.mock('../../../common/ui/FlatTextInput', () => 'FlatTextInput');

describe('modules/auth/screens/RegCompletedDialog', () => {
  let store;
  let auth;

  beforeAll(() => {
    RegisterWithEmail.prototype.getChildContext = function getContext() {
      return { navigator };
    };

    RegisterWithEmail.childContextTypes = {
      navigator: React.PropTypes.object,
    };
  });

  beforeEach(() => {
    store = new Store(config);
    auth = store.auth;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('RegCompletedDialog should render correctly if auth.error does not set', () => {
    const tree = renderer.create(
      <RegisterWithEmail
        auth={auth}
        navigator={navigator}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});