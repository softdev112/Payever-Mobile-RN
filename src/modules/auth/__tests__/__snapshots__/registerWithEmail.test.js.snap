// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/auth/screens/RegCompletedDialog RegCompletedDialog should render correctly if auth.error does not set 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFF",
            "borderBottomColor": "#e1e1e1",
            "borderBottomWidth": 1,
            "elevation": 8,
            "flexDirection": "row",
            "height": 55,
            "justifyContent": "flex-end",
            "marginTop": 15,
            "paddingHorizontal": 18,
            "zIndex": 1000,
          },
          Object {
            "paddingHorizontal": 9,
          },
        ],
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "left": 18,
            "position": "absolute",
            "top": 0,
            "zIndex": 1002,
          },
          Object {
            "left": 9,
          },
        ]
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "opacity": 1,
            "padding": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          source="icon-x-24"
          style={
            Array [
              Object {
                "fontFamily": "payeverIcons",
                "fontSize": 24,
              },
              Array [
                Object {
                  "fontFamily": "payeverIcons",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "height": 24,
                    "width": 24,
                  },
                  Object {
                    "color": "#0084ff",
                  },
                  Array [
                    undefined,
                    Object {
                      "color": "#000",
                    },
                  ],
                ],
              ],
            ]
          }
          touchStyle={
            Object {
              "alignItems": "center",
              "borderColor": "blue",
              "borderWidth": 1,
              "justifyContent": "center",
            }
          }
        >
          î¬“
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#0084ff",
                "fontSize": 16,
                "marginLeft": 2,
              },
              Object {
                "fontSize": 16,
                "fontWeight": "400",
                "marginLeft": 2,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1001,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 18,
                "fontWeight": "300",
                "textAlign": "center",
              },
              Object {},
            ]
          }
        >
          Add New User
        </Text>
      </View>
    </View>
    <Image
      source={1}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": null,
          "justifyContent": "flex-end",
          "paddingLeft": 20,
          "resizeMode": "stretch",
          "width": null,
          "zIndex": 1002,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#0084ff",
                "fontSize": 16,
              },
              undefined,
            ]
          }
        >
          Sign Up
        </Text>
      </View>
    </Image>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "justifyContent": "flex-start",
        "overflow": "hidden",
        "paddingHorizontal": 12,
      }
    }
    scrollEnabled={false}
  >
    <View>
      <FlatTextInput
        inputStyle={
          Object {
            "fontSize": 18,
          }
        }
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="First Name"
        renderValidator={[Function]}
        returnKeyType="next"
        value=""
      />
      <FlatTextInput
        inputStyle={
          Object {
            "fontSize": 18,
          }
        }
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Last Name"
        renderValidator={[Function]}
        returnKeyType="next"
        value=""
      />
      <FlatTextInput
        inputStyle={
          Object {
            "fontSize": 18,
          }
        }
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Email Address"
        renderValidator={[Function]}
        returnKeyType="next"
        value=""
      />
      <FlatTextInput
        inputStyle={
          Object {
            "fontSize": 18,
          }
        }
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Password"
        renderValidator={[Function]}
        returnKeyType="next"
        secureTextEntry={true}
      />
      <FlatTextInput
        inputStyle={
          Object {
            "fontSize": 18,
          }
        }
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Confirm Password"
        renderValidator={[Function]}
        returnKeyType="done"
        secureTextEntry={true}
      />
    </View>
  </RCTScrollView>
</View>
`;
